### DISCUSSION TITLE
How to pass error object in codesandbox

### DISCUSSION THREAD

**Discussion by tuppers360** [react-hook-form/react-hook-form#8880] [General]
Hi just wondering how you could pass the errors object to the Input & Controller so you can style the form element?

https://codesandbox.io/s/build-react-hook-form-controller-component-from-scratch-zl17q?file=/src/App.js

**Comment by Moshyfawn** [comment]
It looks like you've re-created [the `useController` hook](https://react-hook-form.com/api/usecontroller) to me; just saying xD

If you wanna go the "render" route, I'd suggest to make use of the `render props` pattern and pass the errors and other states to the child component that way. [Here's a random article](https://www.codedaily.io/tutorials/Using-Functions-as-Children-and-Render-Props-in-React-Components) to visualize what I'm talking about

**Reply by tuppers360** [reply]
Hi thanks for the reply, the sandbox is an example created by Bill. I think you are correct but I just cant get my head around how that works if you know what I mean :-)
